lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -260,16 +329,23 @@ def should_split_output?
 
       def filename_prompt_args
         default   = action == :dump ? DB_DEFAULT_DUMP_FILE : DB_RESTORE_FILE
-        validator = LOCAL_FILE_VALIDATOR if action == :restore && local_backup?
-        [local_file_prompt, default, validator, "file that exists"]
+        validator = LOCAL_FILE_VALIDATOR if action == :restore && backup_type == LOCAL_FILE
+        prompt = "location of the local restore file" if action == :restore
+        if backup_type == S3_FILE || backup_type == SWIFT_FILE
+          prompt ||= object_store_file_prompt
+          default = File.basename(default)
+        else
+          prompt ||= local_file_prompt
+        end
+        [prompt, default, validator, "file that exists"]
       end
 
       def local_file_prompt
-        if action == :restore
-          "location of the local restore file"
-        else
-          "location to save the #{action} file to"
-        end
+        "location to save the #{action} file to"
```

Comment:
I get adding the change for the below, but this doesn't make sense for `:restore` actions for local files (as this question only gets hit for `:restore` actions for local files.  All other provider types are doing the following (even pulled from your swift changes above):

```ruby
@filename  = just_ask(*filename_prompt_args) { |q| q.readline = false } unless action == :restore
```

Except the `ask_local_file_options`, in which this question makes perfect sense since the rake tasks args for `evm:db:restore:local` differ from `evm:db:restore:remote`.

Reply:
This got addressed.  Resolving.